### PR TITLE
Hide placed cell graphics when not in cell editor tab

### LINKED ISSUE #2269: Hide placed cell graphics in editor when not on the cell editor tab
**Describe the bug**
As you can see, the cell is visible on Report page. Is that working as intended? It looks a little weird.

**Expected behavior**
Maybe it should be hidden outside the Editor page.

**Screenshots**
image

### PR DESCRIPTION
**Pull request by Kasterisk**
**Brief Description of What This PR Does**

Small restructuring of the editor scene tree to allow hiding of spawned nodes and made 3d nodes only be shown in the editor tab.

**Related Issues**

Closes #2269 

**Progress Checklist**

Note: before starting this checklist the issue should be marked as non-draft.

- [x] PR author has checked that this PR works as intended and doesn't
      break existing features:
      https://wiki.revolutionarygamesstudio.com/wiki/Testing_Checklist
      (this is important as to not waste the time of Thrive team
      members reviewing this PR)
- [x] Initial code review passed
- [x] Functionality is confirmed working by another person (see above checklist link)
- [x] Final code review is passed and code conforms to the 
      [styleguide](https://github.com/Revolutionary-Games/Thrive/blob/master/doc/style_guide.md).

Before merging all CI jobs should finish on this PR without errors, if
there are automatically detected style issues they should be fixed by
the PR author.

### CODE REVIEW

File: src/microbe_stage/editor/MicrobeEditor.cs [hunk 1]
```diff
@@ -342,6 +345,21 @@ public bool MicrobePreviewMode
         }
     }
 
+    /// <summary>
+    ///   Controls the visibility of spatial nodes in the editor.
+    /// </summary>
+    public bool WorldEntitiesVisible
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditor.cs:351]: JsonIgnore?

Comment (Kasterisk) [reply src/microbe_stage/editor/MicrobeEditor.cs:351]: Yep, almost forgot.

File: src/microbe_stage/editor/MicrobeEditor.cs [hunk 2]
```diff
@@ -342,6 +345,22 @@ public bool MicrobePreviewMode
         }
     }
 
+    /// <summary>
+    ///   Controls the visibility of spatial nodes in the editor.
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditor.cs:349]: Camera is a spatial node and this doesn't hide that, so, this comment is not technically correct. Should probably reword to be about the cell editor, placed cell visuals and editor arrow etc. being hidden

File: src/microbe_stage/editor/MicrobeEditor.cs [hunk 3]
```diff
@@ -180,6 +181,8 @@ public class MicrobeEditor : NodeWithInput, ILoadableGameState, IGodotEarlyNodeR
 
     private bool microbePreviewMode;
 
+    private bool worldEntitiesVisible = true;
```
Comment (hhyyrylainen) [inline src/microbe_stage/editor/MicrobeEditor.cs:184]: This doesn't seem to be needed? The property could be get only as the set always sets the currently given `value` into the specified nodes, so this storage variable is not actually even used for anything.

**Review by hhyyrylainen — CHANGES_REQUESTED** [review]

File: src/microbe_stage/editor/MicrobeEditor.cs (hunk 2 above)
Comment (Kasterisk) [reply src/microbe_stage/editor/MicrobeEditor.cs:349]: I think `graphical spatial nodes` could be more fitting.

Comment (hhyyrylainen) [reply src/microbe_stage/editor/MicrobeEditor.cs:349]: I'd prefer to mention the cell editor visuals (cell parts) explicitly as I think that gives a better picture as to what this is used for.

**Comment by hhyyrylainen** [comment]
Looks like Microsoft disagrees with this: https://docs.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1044
Perhaps it should be a method then? SetEditorCellVIsibility or soemthing?

**Comment by Kasterisk** [comment]
Changed it to a method.

**Review by hhyyrylainen — APPROVED** [review]

**Pull request merged**